spec/lib/metasploit/cache/module/class/ephemeral/rank_spec.rb

```diff
@@ -0,0 +1,128 @@
+RSpec.describe Metasploit::Cache::Module::Class::Ephemeral::Rank do
+  context '#synchronize' do
+    include_context 'ActiveSupport::TaggedLogging'
+
+    subject(:synchronize) do
+      described_class.synchronize(
+          destination: destination,
+          logger: logger,
+          source: source
+      )
+    end
+
+    let(:destination) {
+      FactoryGirl.build(
+          :metasploit_cache_auxiliary_class,
+          rank: destination_rank
+      ).tap { |direct_class|
+        # Set to nil after build so that record passed to #persist_direct_class must fill in the rank, but the built
+        # template still contains a Rank like a real loading scenario.
+        direct_class.rank = nil
+      }
+    }
+    
+    let(:destination_rank) {
+      FactoryGirl.generate :metasploit_cache_module_rank
+    }
+
+    let(:source) {
+      Class.new.tap { |metasploit_class|
+        metasploit_class.extend Metasploit::Cache::Cacheable
+
+        rank_number = destination_rank.number
+
+        metasploit_class.define_singleton_method(:rank) do
+          rank_number
+        end
+      }
+    }
+
+    context 'with #metasploit_class responds to #rank' do
+      context 'with Metasploit::Cache::Module::Rank seeded' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.